tgui/packages/tgui/components/TimeDisplay.js

```diff
@@ -51,19 +53,32 @@ export class TimeDisplay extends Component {
     clearInterval(this.timer);
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.timing) {
+      return null;
+    }
+    return { value: nextProps.value };
```

Comment:
using this to sync it properly if we receive changed props from parent

if it's timing then it's already syncing via `tick()`